### PR TITLE
Private Events: Ruby on Rails Associations

### PR DESCRIPTION
**Pull request by LoboArkano**
Here, we applied the knowledge gained in the previous projects on working with models, controllers, and associations within the web application. 

**Project milestones**
- [x] Build an application that allows creating and log in a user.
- [x] The user can create an event. 
- [x] Show the list of events created for the user.
- [x] Display upcoming and previous events that the user is an attendee.
- [x] List of all events divided by upcoming and past events.

**General requirements:**

- [x] Add a descriptive README file to your repository
- [x] The README must include instructions on how to use the project on your machine
- [x] Use descriptive commit messages
- [x] Do not commit debugging instructions

**GitFlow:**

- [x] Create a development branch and set it as a default branch in your repository.
- [x] Create a feature-branch branch (give it a descriptive name)
- [x] Use Rubocop to validate your code.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+class EventsController < ApplicationController
+  include EventsHelper
+
+  def new
+    @event = Event.new
+  end
+
+  def create
+    @event = User.find_by(id: session[:user_id]).events.build(event_params)
```
Comment (euqueme) [inline app/controllers/events_controller.rb:9]: ##### EventsController

###### `create` action

* [ ] uses #build method on association, i.e. `current_user.events.build(event_params)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/build_method_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

- Create the current_user variable so you don't have to do this `User.find_by(id: session[:user_id])` several times

File: spec/views/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#   it "signs me in" do
+#     visit '/users/new'
+#     within("#users") do
+#       fill_in 'name', with: 'vader'
+#     end
+#     click_button 'Create User'
+#     expect(page).to have_content 'Success'
+#   end
+# end
```
Comment (euqueme) [inline spec/views/user_spec.rb:35]: - [ ] Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @LoboArkano and @ferbaco86 :lollipop: :balloon: 

### Thank you for your submission :handshake: Your repository is very professional and I can see you've added style to your app and that makes it stand out from the rest :medal_sports: :medal_sports: I also didn't find any logic in views so way to go! :clap: :clap: :clap: There is still some things to work on :construction: :construction_worker_man: :construction_worker_man: 

#### Behaviour

* [x] The event's show page displays the list of attendants **(covered by automated tests)** ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/events_attendees_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))

* Every event has to have a link that takes the user to the event's show page, currently is only possible to see the view by putting it directly in the URL 

* The Create event-form should have an option where guests can be added, currently is only possible in the console

#### Controllers

- [x] I've put comments n your code where changes have to be made 

#### Tests

* [x] models have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
* [x] event managment features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

- I see you have some test cases but they are retrieving errors, please configure Rspec and Capybara properly and add the missing test cases
image

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team!
### You have done a good job so far :clap: However, you have some issues you need to attend to before your project could be approved.
- [x] Validate your form fields so that they don't submit empty records
empty
- [x] RSpec test cases are still failing, you need to fix this
rspec
### Make these changes and submit for another code review. Good luck!
- Happy coding!

By Raymond
slack @iamraymond

File: spec/models/attendance_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    it 'should have many events' do
+      a = Attendance.reflect_on_association(:event_attendee)
+      expect(a.macro).to eq(:belongs_to)
+    end
+    it 'should have many users' do
+      a = Attendance.reflect_on_association(:attended_event)
+      expect(a.macro).to eq(:belongs_to)
+    end
+  end
+end
```
Comment (rOluochKe) [inline spec/models/attendance_spec.rb:14]: ### use require_relative which is the rails generic way of loading files
rspec

**Review by Ispirett — APPROVED** [review]
# Status Approved :1st_place_medal: 
Hey folks great job with all the changes made you did well great :100:  .

**Pull request merged**